# :microscope: Archerdx Code Challenge :microscope:

This particular code challenge was split into three parts <br />
1. Recursively find all FASTQ files in a directory and report each file name and the percent of sequences in that file that are greater than 30 nucleotides long.
2. Given a FASTA file with DNA sequences, find 10 most frequent sequences and return the sequence and their counts in the file.
3. Given a chromosome and coordinates, write a program for looking up its annotation.  Keep in mind you'll be doing this annotation millions of times.
Input:
Tab-delimited file: Chr<tab>Position
GTF formatted file with genome annotations.
Output:
Annotated file of gene name that input position overlaps.
Hint: Most of the sequence reads come from a small portion of the genome. Try to use this information to improve performance, if possible.

The following repo contains a __Ruby__ and a __Python__ (currently under construction) directory that each contain the challenges in their corresponding languages.





### Depndancies
This project is using the follwing
* ruby version 2.5.3



### Install
* clone repo [here](git@github.com:SiCuellar/archerdx_challenge.git)
* cd into the desired code directory __Ruby__ or __Python__
* cd into desired script __numbered above__
* run the __lib/runner.rb__ file and follow the instructions on placing analysis file in the correct location.

### Testing
To run test enter the desired script as instructed above and run the __test/script_name_test.rb__ file

## Personal Information
* LinkedIn [click here](https://www.linkedin.com/in/silvestre-cuellar/)
* Personal Website [click here](https://silvestrecuellar.dev/)(under construction)
* GitHub [click here](https://github.com/SiCuellar)
* Twitter [click here](https://twitter.com/Silver__Codes)
* Turing Portfolio [click here](https://alumni.turing.io/alumni/silvestre-cuellar)


[](https://media.giphy.com/media/dWIL4na2N5OtBe8AqN/giphy.gif)
